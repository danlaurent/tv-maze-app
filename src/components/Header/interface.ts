export interface IHeader {
  onBackPress: () => void;
  testID?: string;
}
