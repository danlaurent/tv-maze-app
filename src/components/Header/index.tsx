import React from 'react';
import { StyleSheet, View, TouchableOpacity, StatusBar } from 'react-native';
import { FontAwesome5 } from '@expo/vector-icons';
import { IHeader } from './interface';

const styles = StyleSheet.create({
  container: {
    alignSelf: 'stretch',
  },
  content: {
    flexDirection: 'row',
    paddingHorizontal: 20,
    paddingVertical: 20,
  },
  statusBar: {
    height: StatusBar.currentHeight,
    alignSelf: 'stretch',
  },
});

const Header = ({ onBackPress }: IHeader) => (
  <View style={styles.container}>
    <View style={styles.statusBar} />
    <View style={styles.content}>
      <TouchableOpacity onPress={onBackPress}>
        <FontAwesome5 name='chevron-left' size={24} color='white' />
      </TouchableOpacity>
    </View>
  </View>
);

export default Header;
