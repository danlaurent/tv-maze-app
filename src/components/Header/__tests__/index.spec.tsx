import React from 'react';
import { render, RenderAPI, fireEvent } from '@testing-library/react-native';
import Header from '..';

const onPressMock = jest.fn();

describe('Header', () => {
  let tree: RenderAPI;

  beforeEach(() => {
    tree = render(
      <Header onBackPress={() => onPressMock()} testID='testHeader' />
    );
  });
  afterEach(() => {
    jest.clearAllMocks();
  });

  it('renders correctly', () => {
    const snapshot = tree.toJSON();
    expect(snapshot).toMatchSnapshot();
  });

  describe('When the back is pressed', () => {
    beforeEach(async () => {
      await fireEvent.press(tree.getByTestId('testHeader_backButton'));
    });

    it('fires the onBackPress action', () => {
      expect(onPressMock).toBeCalledTimes(1);
    });
  });
});
