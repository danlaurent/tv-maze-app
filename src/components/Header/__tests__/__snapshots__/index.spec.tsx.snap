// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<View
  style={
    Object {
      "alignSelf": "stretch",
    }
  }
  testID="testHeader"
>
  <View
    style={
      Object {
        "alignSelf": "stretch",
        "height": null,
      }
    }
  />
  <View
    style={
      Object {
        "flexDirection": "row",
        "paddingHorizontal": 20,
        "paddingVertical": 20,
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="testHeader_backButton"
    >
      <Text />
    </View>
  </View>
</View>
`;
